test/test_distributed.py

```diff
@@ -234,6 +234,20 @@ def test_get_default_group(self):
         default_grp = dist.get_default_group()
         self.assertNotEqual(default_grp, None)
 
+    def test_get_backend(self):
+        if dist.get_world_size() > 2:
+            group = [1, 2]
+        else:
+            group = [0, 1]
+        group_id = dist.new_group(group)
+        backend_val = dist.DistBackend.parse(BACKEND)
+        self.assertEqual(dist.get_backend(), backend_val)
+        if dist.get_rank() in group:
+            self.assertEqual(dist.get_backend(group_id), backend_val)
+        else:
+            with self.assertRaisesRegex(RuntimeError, "Invalid process group specified"):
+                dist.get_backend(group_id)
```

Comment:
@pietern MPI test doesn't seem to like this assertion. Do you have any idea why?

example failure:
```

22:34:07 FAIL: test_get_backend (__main__.TestMPI)
22:34:07 ----------------------------------------------------------------------
22:34:07 ok
22:34:07 Traceback (most recent call last):
22:34:07   File "test_distributed.py", line 249, in test_get_backend
22:34:07     dist.get_backend(group_id)
22:34:07 AssertionError: RuntimeError not raised
22:34:07 
22:34:07 ----------------------------------------------------------------------
22:34:07 Ran 61 tests in 6.023sok
```

Reply:
Must have something to do with MPI group support. I believe this was a challenge for @teng-li so I wouldn't be surprised if there is something related going on here. Looking...